task_geo/data_sources/covid/south_korea/south_korea_patients.py

```diff
@@ -0,0 +1,55 @@
+import io
+
+import pandas as pd
+import requests
+
+
+def south_korea_patients_connector(*args, **kwargs):
+    """Retrieves data from south_korea_patients.
+
+    Arguments:
+        url(string): Dataset url
+    Returns:
+        pandas.DataFrame
+    """
+    csv = requests.get(kwargs['url']).content
+    return pd.read_csv(io.StringIO(csv.decode('utf-8')))
+
+
+def south_korea_patients_formatter(df):
+    """Formats data retrieved from south_korea_patients.
+
+    Arguments:
+      df(pandas.DataFrame):
+
+    Returns:
+      pandas.DataFrame
+    """
+    cols_ordered = [
+        'country', 'state', 'province', 'confirmed_date',
+        'released_date', 'deceased_date', 'exposure_start',
+        'exposure_end', 'global_id', 'birth_year',
+        'local_id', 'sex', 'disease',
+        'group', 'infection_reason', 'infection_order',
+        'infected_by', 'contact_number'
+    ]
+    df = df.reindex(columns=cols_ordered)
+    df['confirmed_date'] = pd.to_datetime(df.confirmed_date)
```

Comment:
This cast can be done in two lines with:
```python
date_columns = [...]
df[date_columns] = pd.to_datetime(df[date_columns])
```

Reply:
Could you elaborate on this?

Reply:
Sure thing.

On the lines between 37-41(I only selected the line 37 to comment, a blunder of mine) you are casting columns to `datetime` and reassigning them multiple times. This approach has two drawbacks:

1. More lines of code to read and write, making it easier to miss details and introduce errors.
2. It's in fact much faster making the casting and assigning of all the columns at once. This is what is called **vectorization** and pandas ( and numpy too) are designed to have vectorized operations run much faster than regular iteration in python.

Reply:
Also, passing the date format to `to_datetime` will further improve its performance.